Lib/typing.py

```diff
@@ -661,10 +661,10 @@ class ForwardRef(_Final, _root=True):
 
     __slots__ = ('__forward_arg__', '__forward_code__',
                  '__forward_evaluated__', '__forward_value__',
-                 '__forward_is_argument__', '__forward_is_class__',
+                 '__forward_is_argument__', '__forward_allow_special_forms__',
                  '__forward_module__')
 
-    def __init__(self, arg, is_argument=True, module=None, *, is_class=False):
+    def __init__(self, arg, is_argument=True, module=None, *, allow_special_forms=False):
```

Comment:
You can't just delete the is_class keyword argument, can you? In the end ForwardRef is a public API.

Reply:
Good point! The last code I touched was the `_ForwardRef` version in older pythons when backporting in `typing_extensions` so I think I had that mindset. Let's leave this `is_class` for now but still change the dunder name, I think.